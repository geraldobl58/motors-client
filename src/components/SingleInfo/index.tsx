import Heading from 'components/Heading'

import { formatPrice } from 'utils/formattedPrice'

import * as S from './styles'

export type SingleInfoProps = {
  make: string
  title: string
  year: string
  milage: number
  fuel: string
  exchange: string
  color: string
  endOfPlate: number
  price: number
  description: string
}

const SingleInfo = ({
  make,
  title,
  year,
  milage,
  fuel,
  exchange,
  color,
  endOfPlate,
  price,
  description
}: SingleInfoProps) => (
  <S.Wrapper>
    <S.Header>
      <Heading color="primary">
        {make} {title}
      </Heading>
      <S.Price>{formatPrice(price)}</S.Price>
    </S.Header>

    <S.Inset>
      <S.Container>
        <S.Title>Ano</S.Title>
        <S.Content>{year}</S.Content>
      </S.Container>
      <S.Container>
        <S.Title>KM</S.Title>
        <S.Content>{milage}</S.Content>
      </S.Container>
      <S.Container>
        <S.Title>Combustível</S.Title>
        <S.Content>{fuel}</S.Content>
      </S.Container>
      <S.Container>
        <S.Title>Câmbio</S.Title>
        <S.Content>{exchange}</S.Content>
      </S.Container>
      <S.Container>
        <S.Title>Cor</S.Title>
        <S.Content>{color}</S.Content>
      </S.Container>
      <S.Container>
        <S.Title>Final de placa</S.Title>
        <S.Content>{endOfPlate}</S.Content>
      </S.Container>
    </S.Inset>

    <S.About>
      <S.Title>Sobre este carro</S.Title>
      <S.Description dangerouslySetInnerHTML={{ __html: description }} />
    </S.About>
  </S.Wrapper>
)

export default SingleInfo
