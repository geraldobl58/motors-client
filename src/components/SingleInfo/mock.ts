export default {
  title: 'Territory',
  year: 2022,
  milage: 2500,
  fuel: 'Gasolina',
  exchange: 'Manual',
  color: 'Branco',
  endOfPlate: 5,
  price: 198900,
  description: 'Mussum Ipsum, cacilds vidis litro abertis.'
}
