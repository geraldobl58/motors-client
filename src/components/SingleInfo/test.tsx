import 'match-media-mock'
import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import SingleInfo from '.'

const props = {
  title: 'Territory',
  year: '2020/2021',
  milage: 2500,
  fuel: 'Gasolina',
  exchange: 'Manual',
  color: 'Branco',
  endOfPlate: 5,
  price: 198900,
  description: 'Mussum Ipsum, cacilds vidis litro abertis.'
}

describe('<SingleInfo />', () => {
  it('should render single information', () => {
    const { container } = renderWithTheme(<SingleInfo {...props} />)

    expect(
      screen.getByRole('heading', { name: /territory/i })
    ).toBeInTheDocument()

    expect(screen.getByText(/2500/i)).toBeInTheDocument()

    expect(screen.getByText(/gasolina/i)).toBeInTheDocument()

    expect(screen.getByText(/manual/i)).toBeInTheDocument()

    expect(screen.getByText(/branco/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
