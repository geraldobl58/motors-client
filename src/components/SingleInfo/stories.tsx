import { Story, Meta } from '@storybook/react/types-6-0'
import SingleInfo, { SingleInfoProps } from '.'

import mock from './mock'

export default {
  title: 'SingleInfo',
  component: SingleInfo,
  args: mock
} as Meta

export const Basic: Story<SingleInfoProps> = (args) => <SingleInfo {...args} />
