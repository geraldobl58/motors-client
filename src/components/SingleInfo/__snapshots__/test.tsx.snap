// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleInfo /> should render single information 1`] = `
.c2 {
  color: #363636;
  font-size: 2.0rem;
}

.c0 {
  padding: 2.4rem 4.8rem;
  border-radius: 0.4rem;
  background: #FAFAFA;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 {
  display: block;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 4.0rem;
}

.c6 {
  color: #7D7D7D;
  font-weight: 300;
  font-size: 1.2rem;
}

.c7 {
  color: #222222;
  font-size: 1.6rem;
}

.c3 {
  font-weight: 700;
  font-size: 2.8rem;
}

.c8 {
  margin-top: 4.0rem;
}

.c9 {
  font-size: 1.8rem;
}

@media (min-width:768px) {
  .c2 {
    font-size: 2.8rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h2
      class="c2"
      color="primary"
    >
      Territory
    </h2>
    <p
      class="c3"
    >
      R$ 198.900,00
    </p>
  </div>
  <div
    class="c4"
  >
    <div
      class="c5"
    >
      <h5
        class="c6"
      >
        Ano
      </h5>
      <h3
        class="c7"
      >
        2020/2021
      </h3>
    </div>
    <div
      class="c5"
    >
      <h5
        class="c6"
      >
        KM
      </h5>
      <h3
        class="c7"
      >
        2500
      </h3>
    </div>
    <div
      class="c5"
    >
      <h5
        class="c6"
      >
        Combustível
      </h5>
      <h3
        class="c7"
      >
        Gasolina
      </h3>
    </div>
    <div
      class="c5"
    >
      <h5
        class="c6"
      >
        Câmbio
      </h5>
      <h3
        class="c7"
      >
        Manual
      </h3>
    </div>
    <div
      class="c5"
    >
      <h5
        class="c6"
      >
        Cor
      </h5>
      <h3
        class="c7"
      >
        Branco
      </h3>
    </div>
    <div
      class="c5"
    >
      <h5
        class="c6"
      >
        Final de placa
      </h5>
      <h3
        class="c7"
      >
        5
      </h3>
    </div>
  </div>
  <div
    class="c8"
  >
    <h5
      class="c6"
    >
      Sobre este carro
    </h5>
    <p
      class="c9"
    >
      Mussum Ipsum, cacilds vidis litro abertis.
    </p>
  </div>
</div>
`;
